### ISSUE TITLE
Login screen doesn't show up when booting with HDMI cable connected

### ISSUE THREAD

**Issue by misiak1989** [nexus511/gpd-ubuntu-packages#22]
Ubuntu 16.04.3 (ubuntu-16.04.3-desktop-amd64_gpdpocket_2017-08-29.iso), updated to gdpocket package version 1.1.

1. When the GPD Pocket has HDMI cable attached when turned off, short pressing the power button shows battery indicator on external screen only, while internal screen remains blank (but that is not relevant to linux itself, as it has not yet started booting even).
2. When powering on like this, the internal screen lights up, but shows nothing - the linux console and mouse cursor for a brief moment get shown on external screen only.
3. When the login screen should show up, the external screen shows "no signal" and internal screen remains blank. Disconnecting HDMI cable does not help.

The GPD Pocket has to be hard-reset (holding power button for few seconds) with HDMI cable disconnected to be usable again (and display login on internal screen).

Attaching pictures demonstrating issue. Sorry for blue rectangled - too much of me and the room got reflected in the black screen, so I had to mask it ;)

![battery indicator on external screen only](https://user-images.githubusercontent.com/31709847/30182939-e10ff6f0-9418-11e7-90dc-91f46fea0625.jpg)
![ubuntu starting up on external screen only](https://user-images.githubusercontent.com/31709847/30182940-e1145c5e-9418-11e7-9b9d-4d0d7886bb4e.jpg)
![no signal on external screen and blank internal screen when login form should be visible](https://user-images.githubusercontent.com/31709847/30182941-e114e8f4-9418-11e7-9613-87acfff91177.jpg)
![detaching hdmi cable doesn't help and the internal screen still remains blank](https://user-images.githubusercontent.com/31709847/30182942-e116223c-9418-11e7-9384-944e16d26b77.jpg)

**Comment by nexus511** [comment]
@misiak1989 I see what you mean and I can reproduce it.

I actually got a login-screen after touching the screen but after login, everything went black and needed a hard-reset.

So booting with HDMI connected is still not a good idea. I think it has been mentioned somewhere before.

**Comment by misiak1989** [comment]
Haven't tried touching the screen, but will... I will also try to ssh into the machine in the meantime to see if dmesg or syslog contain anything useful that could be attached here ;) Is Hans de Goede maintaining the kernel that you patched or did he "just happen to" make a set of some patches that work on our machines? If he is maintaining it, are you forwarding this stuff to him?

**Comment by nexus511** [comment]
In the USB-C issue, @jwrdegoede told us that he is working on the kernel.

Looking at the source and his changes, the basement of the patches has drastically changed and I guess that this is the reason why he will rebase against 4.14.0 and not 4.13.0 once it is out.

I would suggest we wait. I already looked at the source of his latest changes and some of it looks really promising. I would have tried rebasing the changes onto our kernel but for now that looks like a job for several hours of work. So if I find some spare time, I might do this but I actually think that we have to wait for the next kernel version and then check the issues we have here against the new version.

**Comment by jwrdegoede** [comment]
Hi,

About the subject of this issue, which I've just noticed
("Login screen doesn't show up when booting with HDMI cable connected").

I've seen this on the GPD win and I've not tried to reproduce it on the pocket yet,
but if it is happening there that is not too surprising. The BIOS does something
when lighting up the internal panel which the kernel does not do, so the internal
panel does not show a picture. As a workaround if you boot without hdmi connected
and then plug it in once Linux is booting things should work fine. I've looking
into this on my todo list, but no idea when I will get around to this and if
I will be able to fix it.

Regards,

Hans

**Comment by nexus511** [comment]
So I set this (for now) to won't fix and keep it open. We then could wait if it gets fixed at some time in the future by any kernel update.

**Comment by misiak1989** [comment]
I agree this is a minor issue that has a simple workaround... @jwrdegoede , are you also able to reproduce #20 maybe? Is there anything I could do to help? That one is much more serious, as it blocks video and audio playback when trying to use hdmi for both... Are the latest kernel sources on github somewhere? @nexus511 is https://github.com/nexus511/linux-sunxi/ the correct repository to check for kernel modules? Or is the HDMI audio not actually handled by kernel, but by something else (simple search for "lpe" or "hdmi" in that repository finds nothing)?

**Comment by Herst** [comment]
> Are the latest kernel sources on github somewhere?

Hans' can be found at: https://github.com/jwrdegoede/linux-sunxi

**Comment by nexus511** [comment]
@misiak1989  https://github.com/nexus511/linux-sunxi/ is just forked from https://github.com/jwrdegoede/linux-sunxi

So what I do is merge changes from Hans repository from time to time. If you do want to work on the implementation, you want to fork from Hans repository.
